# DataFrame Analysis Exercise

In [1]:
from pathlib import Path

import pandas as pd

In [2]:
# Path to data directory
data_dir = Path.cwd() / "course_materials" / "data"

**Import the `bestsellers.csv` dataset and use it to answer the following questions:**

* Find the lowest User Rating in the DF
* Find the highest Price in the DF
* What is the average User Rating?
* What the average User Rating of the first 5 rows?
* What User Review score appeared the most?
* What is the total (sum) of all the values in the Reviews column?
* How many different authors are featured in the dataset?
* Which author wrote the most number of books on the list?  How many did they write?

In [3]:
# Bestsellers dataset
bestsellers_df = pd.read_csv(data_dir / "bestsellers.csv")

bestsellers_df

Name  \
0                        10-Day Green Smoothie Cleanse   
1                                    11/22/63: A Novel   
2              12 Rules for Life: An Antidote to Chaos   
3                               1984 (Signet Classics)   
4    5,000 Awesome Facts (About Everything!) (Natio...   
..                                                 ...   
545       Wrecking Ball (Diary of a Wimpy Kid Book 14)   
546  You Are a Badass: How to Stop Doubting Your Gr...   
547  You Are a Badass: How to Stop Doubting Your Gr...   
548  You Are a Badass: How to Stop Doubting Your Gr...   
549  You Are a Badass: How to Stop Doubting Your Gr...   

                       Author  User Rating  Reviews  Price  Year        Genre  
0                    JJ Smith          4.7    17350      8  2016  Non Fiction  
1                Stephen King          4.6     2052     22  2011      Fiction  
2          Jordan B. Peterson          4.7    18979     15  2018  Non Fiction  
3               George Orwell          4.7    21424      6  2017      Fiction  
4    National Geographic Kids          4.8     7665     12  2019  Non Fiction  
..                        ...          ...      ...    ...   ...          ...  
545               Jeff Kinney          4.9     9413      8  2019      Fiction  
546               Jen Sincero          4.7    14331      8  2016  Non Fiction  
547               Jen Sincero          4.7    14331      8  2017  Non Fiction  
548               Jen Sincero          4.7    14331      8  2018  Non Fiction  
549               Jen Sincero          4.7    14331      8  2019  Non Fiction  

[550 rows x 7 columns]

#### Find the lowest User Rating in the DF
#### Find the highest Price in the DF
#### What is the average User Rating?

In [4]:
bestsellers_df.describe()

User Rating       Reviews       Price         Year
count   550.000000    550.000000  550.000000   550.000000
mean      4.618364  11953.281818   13.100000  2014.000000
std       0.226980  11731.132017   10.842262     3.165156
min       3.300000     37.000000    0.000000  2009.000000
25%       4.500000   4058.000000    7.000000  2011.000000
50%       4.700000   8580.000000   11.000000  2014.000000
75%       4.800000  17253.250000   16.000000  2017.000000
max       4.900000  87841.000000  105.000000  2019.000000

* The lowest user rating is 3.3.
* The highest price is $105.00.
* The average user rating is 4.62.

#### What the average User Rating of the first 5 rows?

In [5]:
bestsellers_df.head().describe()

User Rating       Reviews      Price         Year
count     5.000000      5.000000   5.000000     5.000000
mean      4.700000  13494.000000  12.600000  2016.200000
std       0.070711   8257.352269   6.308724     3.114482
min       4.600000   2052.000000   6.000000  2011.000000
25%       4.700000   7665.000000   8.000000  2016.000000
50%       4.700000  17350.000000  12.000000  2017.000000
75%       4.700000  18979.000000  15.000000  2018.000000
max       4.800000  21424.000000  22.000000  2019.000000

* The average user rating of the first 5 rows is 4.7.

#### What User Review score appeared the most?

In [6]:
bestsellers_df["User Rating"].mode()

0    4.8
Name: User Rating, dtype: float64

* The user review score that appeared the most is 4.8.

#### What is the total (sum) of all the values in the Reviews column?

In [7]:
bestsellers_df.sum(numeric_only=True)

User Rating       2540.1
Reviews        6574305.0
Price             7205.0
Year           1107700.0
dtype: float64

* The total of all the values in the Reviews column is 6,574,305.

#### How many different authors are featured in the dataset?
#### Which author wrote the most number of books on the list?  How many did they write?

In [8]:
bestsellers_df.describe(include="object")

Name       Author  \
count                                                 550          550   
unique                                                351          248   
top     Publication Manual of the American Psychologic...  Jeff Kinney   
freq                                                   10           12   

              Genre  
count           550  
unique            2  
top     Non Fiction  
freq            310

* There are 248 unique authors in the dataset.
* Jeff Kinney wrote the most number of books in the dataset. He wrote 12 books.